apps/gallery/js/thumbnail_item.js

```diff
@@ -32,25 +33,16 @@ function ThumbnailItem(fileData) {
   var url = URL.createObjectURL(fileData.metadata.thumbnail);
   this.imgNode.src = url;
   this.htmlNode.appendChild(this.imgNode);
-
-  this.localize();
 }
 
-ThumbnailItem.formatter = new Intl.DateTimeFormat(navigator.languages, {
-  hour: 'numeric',
-  minute: 'numeric',
-  month: 'long',
-  day: 'numeric',
-  year: 'numeric',
-});
-
 ThumbnailItem.prototype.localize = function() {
   var date = new Date(this.data.date);
+  var formatter = IntlHelper.get('time-stamp');
 
   var descId = !this.data.metadata.video ?
     'imageDated' : 'videoDated';
 
   navigator.mozL10n.setAttributes(this.imgNode, descId, {
-    timeStamp: ThumbnailItem.formatter.format(date)
+    timeStamp: formatter.format(date)
```

Comment:
That’s right, I am not explicitly specifying IntlHelper.observer('time-stamp’) , which I believe we don’t need as this code is invoked via listener on 'date-group'

Here’s the flow in detail:

IntlHelper data-group observe call below is calling localize method of thumbnailList 
https://github.com/mozilla-b2g/gaia/pull/33156/files#diff-598a8e9249c5971f9c555c58eb7eac6aR35

which iterates and call each ThumbnailDateGroup object localize method
https://github.com/mozilla-b2g/gaia/blob/master/apps/gallery/js/thumbnail_list.js#L112

Inside each ThumbnailDateGroup object localize method we are calling localize on all thumbnail item/ thumbnail.
https://github.com/mozilla-b2g/gaia/pull/33156/files#diff-f597ef3c1ed0182ae894ea21db37491aR142

So here we are using a single listener on date-group to format two formatter objects ‘date-group’ and ’time-stamp’ and it’s appears to be working :) . Do you foresee any issues in using a single observe call here?

Thanks!